src/web/WebSettingsService.cpp

```diff
@@ -88,90 +87,109 @@ void WebSettings::read(WebSettings & settings, JsonObject root) {
+
 #if defined(EMSESP_STANDALONE)
-            settings.board_profile = "S32";
+        settings.board_profile = "S32";
 #elif CONFIG_IDF_TARGET_ESP32
-            if (settings.board_profile == "" && !psram) { // empty: new test for E32
+        // check for no PSRAM, could be a E32 or S32
+        if (!psram) {
 #if ESP_ARDUINO_VERSION_MAJOR < 3
-                if (ETH.begin(1, 16, 23, 18, ETH_PHY_LAN8720, ETH_CLOCK_GPIO0_IN)) {
+            if (ETH.begin(1, 16, 23, 18, ETH_PHY_LAN8720, ETH_CLOCK_GPIO0_IN)) {
 #else
-                if (ETH.begin(ETH_PHY_LAN8720, 1, 23, 18, 16, ETH_CLOCK_GPIO0_IN)) {
+            if (ETH.begin(ETH_PHY_LAN8720, 1, 23, 18, 16, ETH_CLOCK_GPIO0_IN)) {
 #endif
-                    EMSESP::nvs_.putString("boot", "E32");
-                } else {
-                    EMSESP::nvs_.putString("boot", "Test");
-                }
-            } else if (settings.board_profile == "Test" || psram) { // test E32V2
+                settings.board_profile = "E32";
+            }
+        } else {
+// check for PSRAM, could be a E32V2 or S3
 #if ESP_ARDUINO_VERSION_MAJOR < 3
-                if (ETH.begin(0, 15, 23, 18, ETH_PHY_LAN8720, ETH_CLOCK_GPIO0_OUT)) {
+            if (ETH.begin(0, 15, 23, 18, ETH_PHY_LAN8720, ETH_CLOCK_GPIO0_OUT)) {
 #else
-                if (ETH.begin(ETH_PHY_LAN8720, 0, 23, 18, 15, ETH_CLOCK_GPIO0_OUT)) {
+            if (ETH.begin(ETH_PHY_LAN8720, 0, 23, 18, 15, ETH_CLOCK_GPIO0_OUT)) {
 #endif
-                    EMSESP::nvs_.putString("boot", "E32V2");
-                } else {
-                    EMSESP::nvs_.putString("boot", "S32");
-                }
-            } else {
-                EMSESP::nvs_.putString("boot", "S32");
+                settings.board_profile = "E32V2";
             }
-            // ESP.restart();
-            EMSESP::system_.restart_requested(true);
+        }
 #elif CONFIG_IDF_TARGET_ESP32C3
-            settings.board_profile = "C3MINI";
+        settings.board_profile = "C3MINI";
 #elif CONFIG_IDF_TARGET_ESP32S2
-            settings.board_profile = "S2MINI";
+        settings.board_profile = "S2MINI";
 #elif CONFIG_IDF_TARGET_ESP32S3
-            settings.board_profile = "S32S3"; // BBQKees Gateway S3
+        settings.board_profile = "S32S3"; // BBQKees Gateway S3
 #else
-            settings.board_profile = "S32";
+        settings.` = "S32"; // defaulting to an S32
 #endif
-            System::load_board_profile(data, settings.board_profile.c_str());
-        }
-        EMSESP::logger().info("No board profile found. Re-setting to %s", settings.board_profile.c_str());
+
+        // apply the new board profile setting
+        System::load_board_profile(data, settings.board_profile.c_str());
+        EMSESP::logger().warning("No Board profile setup - using %s", settings.board_profile.c_str());
     } else {
-        EMSESP::logger().info("Loading board profile %s", settings.board_profile.c_str());
+        if (old_board_profile != settings.board_profile) {
```

Comment:
Here we need also a ` System::load_board_profile` to initialite the data.
Otherwise crashes with empty FS.

Reply:
don't think its needed, the else is from the other if. its hard to follow with all the #ifdefs.

Reply:
Youre right.